rats-devtools/src/python/rats/annotations/_functions.py

```diff
@@ -60,11 +67,11 @@ def make(self, name: str) -> AnnotationsContainer:
         )
 
 
-AnnotationDecorator = Callable[[T], T]
+DecoratorType = TypeVar("DecoratorType", bound=Callable[..., Any])
 
 
-def annotation(namespace: str, group_id: NamedTuple) -> AnnotationDecorator:
-    def decorator(fn: Callable[..., Any]) -> Callable[..., Any]:
+def annotation(namespace: str, group_id: NamedTuple) -> Callable[[DecoratorType], DecoratorType]:
+    def decorator(fn: DecoratorType) -> DecoratorType:
```

Comment:
ugh ty, this stuff was a nightmare and the reason i ended up ignoring the linting and pushing at 2am after fighting it all night

Reply:
Happy to help! Type hints on decorators are def tricky, especially when you have functions acting as factories for them!